### PR TITLE
Encrypt lbryum wallet (WIP)

### PR DESCRIPTION
**Pull request by kaykurokawa**
This is preliminary work to encrypt lbryum wallet using a password and lbryum's builtin encryption scheme. 

We add new functions is_encrypted and encrypt_wallet to the wallet interface. 

Functions that require decryption take password as argument. 

Will probably need to figure out a way to make it work alongside api sessions pull requests https://github.com/lbryio/lbry/pull/171

### CODE REVIEW

File: lbrynet/core/Wallet.py [hunk 1]
```diff
@@ -1251,12 +1257,12 @@ def check_caught_up():
...
-    def _get_cmd_runner(self):
-        self.cmd_runner = Commands(self.config, self.wallet, self.network)
-
+    def _get_cmd_runner(self, password = None):
+        return Commands(self.config, self.wallet, self.network, password = password)
+        
     def get_balance(self):
         cmd = known_commands['getbalance']
-        func = getattr(self.cmd_runner, cmd.name)
+        func = getattr(self._get_cmd_runner(), cmd.name)
```
Comment (jobevers) [inline lbrynet/core/Wallet.py]: can this be made into a function, like

```
def get_command(name, password=None):
    cmd = known_commands(name)
    return getattr(self._get_cmd_runner(password), cmd.name)
```

File: lbrynet/core/Wallet.py [hunk 2]
```diff
@@ -448,7 +449,7 @@ def update_metadata(self, new_metadata, old_metadata):
             meta_for_return[k] = new_metadata[k]
         return defer.succeed(Metadata(meta_for_return))
 
-    def claim_name(self, name, bid, m):
+    def claim_name(self, name, bid, m, password = None):
```
Comment (jobevers) [inline lbrynet/core/Wallet.py]: style nit: no spaces around the `=`. It should be: `password=None`

**Comment by jobevers** [comment]
Made a couple of small comments / suggestions.  Is this still "WIP"?  I think it could be merged in when those bits get cleaned-up.

**Comment by kaykurokawa** [comment]
Will review , test and merge

**Comment by lyoshenka** [comment]
@jobevers can we get this on the list for next cycle?

**Comment by kaykurokawa** [comment]
I won't have time to work on this for next cycle. This also requires integration on the daemon and UI side for encryption to work ( although merging it in will not break anything )

**Comment by jobevers** [comment]
Getting the PR merged, but not doing the rest of the daemon integration and
UI work, would be useful.

**Comment by kaykurokawa** [comment]
Closing this as some upcoming changes to the wallet will mean that this will need to be rewritten.

**Comment by dskvr** [comment]
:|

**Comment by tzarebczan** [comment]
We had some recent work done on encryption by another Dev. Definitely on
our list and a wip.

**Pull request closed without merging**